### ISSUE TITLE
Improve latest update processing for Referenceables

### ISSUE THREAD

**Issue by mandy-chessell** [odpi/egeria#3308]
Area 0 allows the OMASs to maintain an audit trail of changes to referenceable and record the latest change in an asset.

![0010-Base-Model](https://user-images.githubusercontent.com/12354195/86781611-06addf00-c056-11ea-9044-91d70325695a.png)

Now that classifications can be stored independently of an entity we can improve the management of this information by using a classification to store the information.  This will remove the need to locate and update the last attachment entity to record any attachement changes.  It also means the latest changes to the Asset can be maintained even if the Asset is owned by an external technology and can not be updated directly.

This is the new proposed 0010 model:

![0010-Base-Model](https://user-images.githubusercontent.com/12354195/86786091-7d99a680-c05b-11ea-86ba-737b85f225ee.png)

Maintaining this change information makes the event processing in some of the OMASs that create notifications when an asset (and its many types of attachments) change because it just need to monitor this new classification rather than all types of entities and relationships that can be connected to the asset.

**Comment by mandy-chessell** [comment]
Since writing the above I have discovered that we can further improve this processing.  The **LatestChange** classification is attached to an anchor Referenceable.  So for example, if a hierarchy of SchemaElements, or a hierarchy of Comments (all Referenceables) were attached to an Asset (also a Referenceable), then the LatestChange classification goes on the Asset and records changes to any of these elements. 

Maintaining "LatestChanges" on Asset means that as any of these connected elements are updated, deleted or have things attached to them, the Asset entity needs to be efficiently located to add details of the change to its classification.

We had been adding a new property called `assetGUID` to types such as SchemaElement and Comment to make it easier to find the anchor entity.  This is working but the downside of this approach is:
* This property seems out of place and difficult to explain.
* it is not possible to update it if the entity is owned by an external thrid party technology as the entity properties are effectively readonly.

Therefore I propose to add a new classification called **Anchors** that is attached to any entity with a lifecycle dependency on an anchor Referenceable.  In the example above, Anchors would be on all of the SchemaElements and Comments.  Anchors would contain the unique identifer (guid) of the anchor Referenceable.  I have also added the root of a schema structure and the root of a comment tree to the anchors classification to make it easier to process these elements too.

The picture looks like this ...

![anchors-classifications-on-dependant-objects](https://user-images.githubusercontent.com/12354195/89044954-9dfb0f00-d342-11ea-9671-e9c039ef7344.png)

The entities that have the Anchors classification are those that are deleted if the anchor is deleted.  Entities such as Ratings, Likes and Attachments (from ODF) do have a lifecycle that is dependent on their anchors and so would include an Anchors classification.

If, for example, a GlossaryTerm, or InformalTag is attached, since they potentially outlive the delete of the Asset then they do not have an Anchors classification.  Any change to these entities does not reflect in the LatestChange classification of the Asset.  However, the act of attaching them, or detaching them is to the Asset is recorded in the Asset's LatestChange.

This approach also has additional value when it comes to NoteLogs.  These are Referenceables that can be attached to many other Referenceables and in that case they would have an independent lifecycle (and hence no Anchors classification).  However there a number of use cases where the NoteLog is connected to only one Referenceable and it shoul be linked both to that referenceable's lifecycle and visibility.  For example:
* Notelogs are used to support the personal Blog linked off of the PersoanlProfile in CommunityProfile OMAS.
* Assets may have a notelog to record "news" for consumers such as planned maintenance and unexpected situations.

We can use the Anchors classification on a NoteLog to say that the noteLog is tied to the referenceable it is attached to.  This would then prevent a new relationship being added to link it to another referenceable.  Then it could be bound both by
its linked referenceable's visibility and lifecycle.

Below is the updated UML showing the Anchors classification.  I moved the management type attachments for Referenceables to a new sheet as 0010 was getting crowded.

![0010-Base-Model](https://user-images.githubusercontent.com/12354195/89052554-c9372b80-d34d-11ea-995b-0979f49460a7.png)
![0011-Managing-Referenceables](https://user-images.githubusercontent.com/12354195/89052560-cc321c00-d34d-11ea-8ed7-53a94970c164.png)
